src/backend/core/migrations/0006_merge_duplicate_users.py

```diff
@@ -0,0 +1,64 @@
+
+from django.db import migrations, models
+from django.db.models import Count
+
+def merge_duplicate_user_accounts(apps, schema_editor):
+    """Merge user accounts that share the same email address.
+
+    Historical Context:
+    Previously, ProConnect authentication could return users with the same email
+    but different sub, leading to duplicate user accounts. While the application
+    now prevents this scenario, this migration is needed to clean up existing
+    duplicate accounts to ensure users can continue to connect without being blocked
+    by unique email constraints.
+    """
+
+    User = apps.get_model('core', 'User')
+    ResourceAccess = apps.get_model('core', 'ResourceAccess')
+
+    emails_with_duplicates = (
+        User.objects.values('email')
+        .annotate(count=Count('id'))
+        .filter(count__gt=1)
+        .values_list('email', flat=True)
+    )
+
+    for email in emails_with_duplicates:
+        # Keep the oldest user
+        primary_user = User.objects.filter(email=email).order_by('created_at').first()
+        duplicate_user_accounts = User.objects.filter(email=email).exclude(id=primary_user.id)
+
+        # Get IDs of duplicate accounts to be merged
+        duplicate_account_ids = list(duplicate_user_accounts.values_list('id', flat=True))
+        resource_accesses_to_transfer = ResourceAccess.objects.filter(user_id__in=duplicate_account_ids)
+
+        # Transfer resource access permissions to primary user
+        for resource_access in resource_accesses_to_transfer:
+
+            # Skip if primary user already has access to this resource
+            existing_primary_access = ResourceAccess.objects.filter(
+                user_id=primary_user.id,
+                resource_id=resource_access.resource.id
+            )
+
+            if existing_primary_access.exists():
+                continue
+
+            # Transfer access to primary user
+            resource_access.user_id = primary_user.id
+            resource_access.save()
```

Comment:
Two points:
 - I would also check to give the "max" access if it already exists (ie if the `existing_primary_access` is lower than the `resource_access` it should be updated, don't you think?
 - I guess all the existing_primary_access could be fetched before the loop.

The second point is not mandatory, as stated before, I guess there are not many updates do the migration will remain fast enough.

Reply:
I fixed your first point !